# Source: https://gitlab.com/gitlab-org/gitlab-foss/-/blob/v18.11.12/Gemfile
# CVE-2025-67202 requires sidekiq-cron 2.4.0: https://nvd.nist.gov/vuln/detail/CVE-2025-67202
# GHSA-3h96-34p3-xm76: graphql 2.6.10 in upstream v18.11.12 satisfies the advisory floor (2.5.26)
# CVE-2026-63435 requires mail 2.9.1: https://nvd.nist.gov/vuln/detail/CVE-2026-63435
# sidekiq-cron 2.4 compatibility: https://github.com/sidekiq-cron/sidekiq-cron/compare/v2.3.1...v2.4.0
diff --git a/Gemfile b/Gemfile
--- a/Gemfile
+++ b/Gemfile
@@ -297,7 +297,7 @@
 
 # Background jobs
 gem 'sidekiq', path: 'vendor/gems/sidekiq', require: 'sidekiq', feature_category: :scalability
-gem 'sidekiq-cron', '~> 2.3.0', feature_category: :scalability
+gem 'sidekiq-cron', '~> 2.4.0', feature_category: :scalability
 gem 'gitlab-sidekiq-fetcher',
   path: 'vendor/gems/sidekiq-reliable-fetch',
   require: 'sidekiq-reliable-fetch',
@@ -687,7 +687,7 @@
 # `config/initializers/mail_starttls_patch.rb` has also been patched to
 # fix STARTTLS handling until https://github.com/mikel/mail/pull/1536 is
 # released.
-gem 'mail', '= 2.9.0', feature_category: :shared # rubocop:todo Gemfile/MissingFeatureCategory -- https://gitlab.com/gitlab-org/gitlab/-/issues/581839
+gem 'mail', '= 2.9.1', feature_category: :shared # rubocop:todo Gemfile/MissingFeatureCategory -- https://gitlab.com/gitlab-org/gitlab/-/issues/581839
 gem 'mail-smtp_pool', '~> 0.1.0', path: 'gems/mail-smtp_pool', require: false, feature_category: :shared # rubocop:todo Gemfile/MissingFeatureCategory -- https://gitlab.com/gitlab-org/gitlab/-/issues/581839
 
 gem 'microsoft_graph_mailer', '~> 0.1.0', path: 'vendor/gems/microsoft_graph_mailer', feature_category: :shared # rubocop:todo Gemfile/MissingFeatureCategory -- https://gitlab.com/gitlab-org/gitlab/-/issues/581839
diff --git a/lib/gitlab/patch/sidekiq_cron_poller.rb b/lib/gitlab/patch/sidekiq_cron_poller.rb
--- a/lib/gitlab/patch/sidekiq_cron_poller.rb
+++ b/lib/gitlab/patch/sidekiq_cron_poller.rb
@@ -11,7 +11,7 @@
   raise 'New version of sidekiq detected, please remove or update this patch'
 end
 
-if Gem::Version.new(Sidekiq::Cron::VERSION) != Gem::Version.new('2.3.1')
+if Gem::Version.new(Sidekiq::Cron::VERSION) != Gem::Version.new('2.4.0')
   raise 'New version of sidekiq-cron detected, please remove or update this patch'
 end
 
